fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraSession.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api;
+
+import static java.time.Duration.ofMinutes;
+import static java.time.Duration.ofMillis;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The Fedora Session abstraction
+ *
+ * @author acoburn
+ */
+public interface FedoraSession {
+
+    // The default timeout is 3 minutes
+    public static final Duration DEFAULT_TIMEOUT = ofMinutes(3);
+
+    public static final String TIMEOUT_SYSTEM_PROPERTY = "fcrepo.transactions.timeout";
+
+    public interface State {}
+
+    public enum RequiredState implements State {
```

Comment:
`ACTIVE` is something that is used in the current impl to test `jcr.Session::isLive`. `BATCH` is a marker to indicate that a particular session persists across multiple HTTP requests.